crates/biome_js_analyze/src/lint/nursery/use_unique_element_ids.rs

```diff
@@ -3,14 +3,164 @@ use biome_console::markup;
 
+/// SVG element tags where `id` attributes have local scope within the SVG
+const SVG_TAGS: [&str; 82] = [
+    "a",
+    "altGlyph",
+    "altGlyphDef",
+    "altGlyphItem",
+    "animate",
+    "animateColor",
+    "animateMotion",
+    "animateTransform",
+    "circle",
+    "clipPath",
+    "color-profile",
+    "cursor",
+    "defs",
+    "desc",
+    "ellipse",
+    "feBlend",
+    "feColorMatrix",
+    "feComponentTransfer",
+    "feComposite",
+    "feConvolveMatrix",
+    "feDiffuseLighting",
+    "feDisplacementMap",
+    "feDistantLight",
+    "feDropShadow",
+    "feFlood",
+    "feFuncA",
+    "feFuncB",
+    "feFuncG",
+    "feFuncR",
+    "feGaussianBlur",
+    "feImage",
+    "feMerge",
+    "feMergeNode",
+    "feMorphology",
+    "feOffset",
+    "fePointLight",
+    "feSpecularLighting",
+    "feSpotLight",
+    "feTile",
+    "feTurbulence",
+    "filter",
+    "font",
+    "font-face",
+    "font-face-format",
+    "font-face-name",
+    "font-face-src",
+    "font-face-uri",
+    "foreignObject",
+    "g",
+    "glyph",
+    "glyphRef",
+    "hatch",
+    "hatchpath",
+    "hkern",
+    "image",
+    "line",
+    "linearGradient",
+    "marker",
+    "mask",
+    "metadata",
+    "missing-glyph",
+    "mpath",
+    "path",
+    "pattern",
+    "polygon",
+    "polyline",
+    "radialGradient",
+    "rect",
+    "script",
+    "set",
+    "stop",
+    "style",
+    "svg",
+    "switch",
+    "symbol",
+    "text",
+    "textPath",
+    "title",
+    "tspan",
+    "use",
+    "view",
+    "vkern",
+];
+
+/// Checks if the given element name is an SVG tag
+fn is_svg_element(element_name: &str) -> bool {
+    SVG_TAGS.binary_search(&element_name).is_ok()
+}
+
+/// Checks if an element is an SVG element or is nested within an SVG context
+fn is_in_svg_context(element: &AnyJsxElement) -> bool {
+    // Check if current element is SVG
+    if let Ok(name) = element.name() {
+        if let Some(element_name) = get_element_name(&name) {
+            if is_svg_element(&element_name) {
+                return true;
+            }
+        }
+    }
+
+    // Walk up the tree to find if we're inside an SVG element
+    let mut current = element.syntax().parent();
+    while let Some(parent) = current {
+        // Check if parent is a JSX element
+        if let Some(jsx_element) = AnyJsxElement::cast(parent.clone()) {
+            if let Ok(name) = jsx_element.name() {
+                if let Some(element_name) = get_element_name(&name) {
+                    if is_svg_element(&element_name) {
+                        return true;
+                    }
+                }
+            }
+        }
+        current = parent.parent();
+    }
+
+    false
+}
+
+/// Extracts the element name from AnyJsxElementName
+fn get_element_name(name: &AnyJsxElementName) -> Option<String> {
+    match name {
+        AnyJsxElementName::JsxName(jsx_name) => {
+            jsx_name.value_token().ok().map(|token| token.text_trimmed().to_string())
```

Comment:
All the calls to `.to_string()` allocates a new string. You should return a `&str` or a `TokenText` instead.